### ISSUE TITLE
highlight followed hashtags

### ISSUE THREAD

**Issue by alltheseas** [damus-io/damus#2088]
## user story

As a Damus hashtag peruser who follows some (and not all) hashtags, when I see a note with multiple hashtags I would like Damus to highlight the hashtags I follow, so that I can differentiate between hashtags I follow, and hashtags I don't follow.

### acceptance criteria

Given I follow one or more hashtags, AND
I see one or more hashtags that I follow in a note in Damus

1. Damus highlights the hashtags I follow (and conversely, does not highlight the hashtags I do not follow)

**Comment by karnagebitcoin** [comment]
@alltheseas how about color difference? Right now all hashtags are link colors. What if the hashtags you don't follow were more muted? Perhaps a grey link? Just a thought.

**Comment by alltheseas** [comment]
Agree with 

1) differentiating followed, not followed, muted hashtags

2) bringing hashtags to the top

**Comment by alltheseas** [comment]
Ig ![image](https://github.com/damus-io/damus/assets/64376233/007ff4a0-3d33-425b-8f46-2db9efcf5636)

**Comment by karnagebitcoin** [comment]
@alltheseas what does bringing to the top do? Just move all the hashtags to the top, or only for those you're following? a bit confused by this as I don't use other social networks

**Comment by karnagebitcoin** [comment]
Ah ok. So maybe show the hashtag on top if you're following it? and if it shows up under say... 4 hashtags all related to photography, you would show one, two or max 3 hashtags (to be determined)?

**Comment by alltheseas** [comment]
1) hashtag(s) should be surfaced on top, irregardless if you are following
2) followed hashtags should be displayed first
3) followed hashtags should be highlighted in "header", and in note

Yes display maybe up to three hashtags. I know real estate is limited. Hashtags should not be greater than one row.

**Comment by karnagebitcoin** [comment]
How far off am I here? All tags at the top. Link color = followed, muted = not followed. <img width="1280" alt="tags" src="https://github.com/damus-io/damus/assets/131526701/db629df1-4030-4926-8c91-7a1a0d3cd6fc">

**Comment by alltheseas** [comment]
I like the header.

I did not think about not showing hashtags in the body. 

What happens if there are 5 hashtags in a note?

**Comment by jb55** [comment]
On Tue, Apr 02, 2024 at 07:15:53AM -0700, Karnage wrote:
>How far off am I here? All tags at the top. Link color = followed, muted = not followed. <img width="1280" alt="tags" src="https://github.com/damus-io/damus/assets/131526701/db629df1-4030-4926-8c91-7a1a0d3cd6fc">

We only need to show followed hashtag in the header, any of them. it
doesn't need to be all of them or multiple. Non-followed will already be
in the body. If they aren't in the body then it doesn't matter, because
they are not related to the reason why you are seeing it in your feed.

The reason we want to show this because if a hashtag note appears in
your home timeline there is no indiciation why its there. Especially if
the post uses lots of hashtags, you want to know which hashtag is
causing this note to appear in your feed.

**Comment by karnagebitcoin** [comment]
Some variations here, hashtag on top, below name, and muted tags in body. 

<img width="1280" alt="variations" src="https://github.com/damus-io/damus/assets/131526701/c92510a7-4bdd-4e1d-84df-8aa9d7be4510">

**Comment by alltheseas** [comment]
3 > 2 > 1

in 3 greyed hashtags make me think muted color means a muted hashtag.

**Comment by karnagebitcoin** [comment]
So maybe 2 then? Could make the followed hashtag more bold too

**Comment by alltheseas** [comment]
Yeah I think distinction between followed and non-followed hashtag is useful

**Comment by karnagebitcoin** [comment]
I can try using the official Figma files if there is one. The file I looked at seems outdated (design looks very different)

![image](https://github.com/damus-io/damus/assets/131526701/6dcdecdb-3f86-4d36-9cd5-232ce69f77aa)

**Comment by alltheseas** [comment]
related https://github.com/damus-io/damus/issues/1848

**Comment by alltheseas** [comment]
@ericholguin 👀

**Comment by alltheseas** [comment]
Closing as part of a backlog triage — we're narrowing focus to the most actionable issues to reduce noise for contributors.

If this is still relevant, please reopen with updated context or reproduction steps. Thanks for reporting!
